src/traffic_layout/engine.h

```diff
@@ -74,6 +75,8 @@ struct RunrootEngine {
   // for parsing
   int command_num = 0;
 
+  std::string layout_file = "";
```

Comment:
The constructor is pointless, since `std::string` default constructs to empty string.